runtime/core/named_data_map.h

```diff
@@ -7,9 +7,6 @@
  */
 
 #pragma once
-// Disable -Wdeprecated-declarations, as some builds use 'Werror'.
-#pragma GCC diagnostic push
```

Comment:
wrap these with

```
#ifdef __GNUC__
```

and I think there will not be any need to remove these lines.